simplemonitor/Monitors/remote_ssh.py

```diff
@@ -0,0 +1,135 @@
+        # description
+        self.description = cast(str, self.get_config_option("description", required=False))  # maybe define default here instead of a try: ?
+        self.success_message = cast(str, self.get_config_option("success_message", required=False, default="it worked"))
+        # ssh configuration
+        self.command = cast(str, self.get_config_option("command", required=True))
+        self.ssh_private_key_path = cast(str, self.get_config_option("ssh_private_key_path", required=True))
+        self.ssh_username = cast(str, self.get_config_option("ssh_username", required=True))
+        self.target_hostname = cast(str, self.get_config_option("target_hostname", required=True))
+        self.target_port = cast(int, self.get_config_option("target_port", required=False, default="22"))
+        # operator logic
+        self.operator = cast(
+            str,
+            self.get_config_option(
+                "operator",
+                required=True,
+                allowed_values=[Operator.EQUALS.value, Operator.NOT_EQUALS.value, Operator.LESS_THAN.value, Operator.GREATER_THAN.value],
+            ),
+        )
+        self.regex = re.compile(cast(str, self.get_config_option("regex", required=True)))
+        # values to compare, cast to the expected type
+        self.result_type = cast(
+            str,
+            self.get_config_option("result_type", required=True, allowed_values=[OperatorType.INTEGER.value, OperatorType.STRING.value]),
+        )
+        match self.result_type:
+            case OperatorType.INTEGER.value:
+                self.target_value = cast(int, self.get_config_option("target_value", required=True))
+            case OperatorType.STRING.value:
+                self.target_value = cast(str, self.get_config_option("target_value", required=True))
+
+    def run_test(self) -> bool:
+        # run remote command
+        client = paramiko.SSHClient()
+        client.set_missing_host_key_policy(paramiko.AutoAddPolicy)
+        try:
+            client.connect(
+                self.target_hostname,
+                username=self.ssh_username,
+                key_filename=self.ssh_private_key_path,
+                port=self.target_port,
+            )
+        except TimeoutError:
+            return self.record_fail(f"connection to {self.target_hostname} timed out")
+        except ConnectionRefusedError:
+            return self.record_fail(f"connection to {self.target_hostname} actively refused")
+        except Exception as e:
+            return self.record_fail(f"connection to {self.target_hostname} failed: {e}")
+        else:
+            _, stdout, _ = client.exec_command(self.command)
+
+        # extract and cast the actual value
+        command_result = stdout.read().decode("utf-8")  # let's hope for the best
+        client.close()  # it's only now that we are done with the client
+        actual_value = re.match(self.regex, command_result).groups()[0]
```

Comment:
I was wondering if it would be better to require a named group in the regexp, so that can be picked out specifically rather than requiring the user to craft a regexp which makes the first capturing group the one they need?

Reply:
I can do that but what would be the value of several capturing groups? There is just one that will be used anyway.